app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeViewModel.kt

```diff
@@ -39,9 +39,9 @@ interface Output : BaseOutput {
 
     val trendingCoins: StateFlow<List<CoinItemUiModel>>
 
-    val myCoinsError: SharedFlow<Throwable?>
+    val myCoinsError: StateFlow<Throwable?>
```

Comment:
`errors` should be `SharedFlow` type or they will be re-emitted again on configuration changes.

Reply:
@kaungkhantsoe @Wadeewee 
- We need to assert that a Toast shows on the screen when an error occurs.
- There is a known issue for now (or at least I can not resolve it yet): Robolectric can not assert a Toast by using `ShadowToast` if we show it by using `SharedFlow`.

So 
- I really expect to have test code to assert the Toast.
- For @kaungkhantsoe concern, we can create a method in BaseViewModel to set the error to null after showing. 🤓

Reply:
Oh, @luongvo @kaungkhantsoe, Actually, I just changed this to make it consistent with what type that we return in `VM` 
<img width="619" alt="Screen Shot 2566-02-17 at 17 45 51" src="https://user-images.githubusercontent.com/28002315/219623674-52cc45d7-551a-41ec-a27f-f2486808a9d0.png">

Reply:
> I really expect to have test code to assert the Toast.

@luongvo Any ideas to `assert` this without `roborectric` 🙏

Reply:
I am still thinking about should we modify the production code to that length to be able to test the toast got shown or not. 🤔

Reply:
@Wadeewee AFAIK, We can not assert a Toast with Jetpack Comose UI test. So:
- We need to use Robolectric to assert the Toast, which requires us to use StateFlow.
- Or switch to using SnackBar instead of Toast, then we can assert by using `onNodeWithText("unexpected error").assertIsDisplayed()`. However, SnackBar implementation requires to use StateFlow too 🤓

Reply:
> production code to that length to be able to test the toast got shown or not.

Sorry @kaungkhantsoe, What did you mean by the `production code`? Could you please explain me more 🙏

Reply:
> We need to use Robolectric to assert the Toast, which requires us to use StateFlow.

Yeah, @luongvo 🙈 That's why I didn't put any assertion for toast in UI Test 😭

Reply:
@Wadeewee but we need that assertion to complete the test :(

Reply:
Finally, @luongvo 😂  I found a way how to assert the `toast` in case we didn't put in any node in the `Composable` function,  So we need to call from the `onRoot()` instead of find on the node:

```kotlin
 composeAndroidTestRule.onRoot().assertExists("message")
 ```
 
 I also update in `HomeScreenTest` as well. In this way, we don't need to use `SadowToast.getTextOfLatestToast()` in Unit Test 🚀

Reply:
>> production code to that length to be able to test the toast got shown or not.
>> 
> Sorry @kaungkhantsoe, What did you mean by the production code? Could you please explain me more 🙏

Regarding this, Now I get @kaungkhantsoe point. His concern is all about changing `SharedFlow to StateFlow` to just testing the toast. this's a good point to raise up. As we know by using `MutableSharedFlow` it doesn't emit the most recent value to the collector, but It will emit only when the collector was created. So It might be that the toast will not display because it was emitted before the collector was created. 

So let me recap 🙏  Because of the reason above ☝️ that's why we need to use `StateFlow` for this when we have `Android Test`, So we can see the toast on the device when running the test case. IMO: For this case, I don't think it's not bad to modify the flow to be able to test in the engineering showcase project because, in the client project, we don't recommend having an `Android Test` for sure. Instead, we will make them to `UnitTest` by using `Robolectric`, and by  using `Robolectric`, we won't face this issue and don't need to modify the flow as it works fine 🙇‍♀️ 

What do you think? cc @luongvo @hoangnguyen92dn

Reply:
Sorry guys 🙏 I thought this assertion was work

```kotlin
 composeAndroidTestRule.onRoot().assertExists("message")
```

So I think we might need `StateFlow` when using `Robolectric` as well. Let's try to investigate later on a new PR may be on the PR for changing `Toast` to `SnackBar` 🙇‍♀️

Reply:
> in the client project, we don't recommend having an Android Test for sure.

@Wadeewee It's the trade-off if we want to have the end-to-end test. By saying "we don't recommend" doesn't mean we should not dive depth into this part 🙏 

Since you decided to leave the Toast validation as TODO, should we revert this change as well?

Reply:
@hoangnguyen92dn That's why I said :

> That's why we need to use StateFlow for this when we have Android Test, So we can see the toast on the device when running the test case. IMO: For this case, I don't think it's bad to modify the flow to be able to test in the engineering showcase project

Reply:
@hoangnguyen92dn, Not sure if you mean for having `Android Test` or modifier the `flow`, but if you mean modifier the flow, I think we still need it when using `Robolectric` as well. I have said this:
 

> in the client project, we don't recommend having an Android Test for sure. Instead, we will make them to UnitTest by using Robolectric, and by using Robolectric, we won't face this issue and don't need to modify the flow as it works fine

Because I think `composeAndroidTestRule.onRoot().assertExists("message")`, will work, but it's not 🙏 

Currently, We have methods to reset the value to null. In this way, I think we can use `StateFlow` 
more safely 🙇‍♀️